datadog_checks_base/datadog_checks/base/checks/windows/perf_counters/counter.py

```diff
@@ -0,0 +1,516 @@
+            # Some instances may not have a value yet, so we only
+            # use the ones that do for computing averages
+            instance_handles_with_data = len(counter_handles)
+            instance_total = 0
+
+            for i, counter_handle in enumerate(counter_handles):
+                try:
+                    value = get_counter_value(counter_handle)
+                except pywintypes.error as error:
+                    instance_handles_with_data -= 1
+                    self.handle_counter_value_error(error, instance if i == 0 else f'{instance}#{i}')
+                else:
+                    instance_total += value
+
+            if not instance_handles_with_data:
+                continue
+
+            handles_with_data += instance_handles_with_data
+            total += instance_total
+
+            if self.aggregate != 'only':
+                tags = [f'{self.tag_name}:{instance}']
+                tags.extend(self.tags)
+
+                if self.average:
+                    self.transformer(instance_total / instance_handles_with_data, tags=tags)
+                else:
+                    self.transformer(instance_total, tags=tags)
+
+        if not handles_with_data:
+            return
+
+        if self.aggregate is not False:
+            if self.average:
+                self.aggregate_transformer(total / handles_with_data, tags=self.tags)
+            else:
+                self.aggregate_transformer(total, tags=self.tags)
+
+    def refresh(self, instance_counts):
+        old_instances = self.instances
+        new_instances = {}
+
+        for instance, current_count in instance_counts.items():
+            if instance in old_instances:
+                counter_handles = old_instances.pop(instance)
+                new_instances[instance] = counter_handles
+                old_count = len(counter_handles)
+
+                if current_count > old_count:
+                    for index in range(old_count, current_count):
+                        path = construct_counter_path(
+                            machine_name=self.connection.server,
+                            object_name=self.object_name,
+                            counter_name=self.name,
+                            instance_name=instance,
+                            instance_index=index,
+                        )
+                        counter_handle = self.counter_selector(self.connection.query_handle, path)
+                        counter_handles.append(counter_handle)
+                elif current_count < old_count:
+                    for _ in range(old_count - current_count):
+                        counter_handle = counter_handles.pop()
+
+                        # https://docs.microsoft.com/en-us/windows/win32/api/pdh/nf-pdh-pdhremovecounter
+                        # http://timgolden.me.uk/pywin32-docs/win32pdh__RemoveCounter_meth.html
+                        win32pdh.RemoveCounter(counter_handle)
```

Comment:
should we log when we remove counters during the refresh?

Reply:
I think not b/c we'd then have to compute a diff of instances for every counter (note we call `.clear` always)

Reply:
https://github.com/DataDog/integrations-core/pull/10504/commits/c0b2dd51b9c4ab486358947b75d20e85d005cc72